libs/sr/solver/src/solver.ts

```diff
@@ -121,48 +125,119 @@ export class Solver {
-         * TODO: make lightcone node detachment opt-in.
-         */
-        if (tag['src'] !== this.characterKey) return undefined // Wrong member
-        if (tag['et'] !== 'own') return undefined // Not applied (only) to self
-
-        if (tag['sheet'] === 'dyn' && tag['qt'] === 'premod')
-          return { q: tag['q']! } // Relic stat bonus
-        if (tag['q'] === 'count' && relicSetKeys.has(tag['sheet'] as any))
-          return { q: tag['sheet']! } // Relic set counter
-        if (
-          tag['qt'] == 'lightCone' &&
-          ['lvl', 'ascension', 'superimpose'].includes(tag['q'] as string)
         )
-          return { q: tag['q']! } // Light cone bonus
-        if (tag['q'] === 'count' && lightConeKeys.has(tag['sheet'] as any))
-          return { q: tag['sheet']! } // Relic set counter
+      ),
+    ]
+    const detachedNodes = detach(nodes, this.calc, (tag: Tag) => {
+      /**
+       * Removes relic and lightcone nodes from the opt character, while retaining data from the rest of the team.
+       * TODO: make lightcone node detachment opt-in.
+       */
+      if (tag['src'] !== this.characterKey) return undefined // Wrong member
+      if (tag['et'] !== 'own') return undefined // Not applied (only) to self
 
-        return undefined
-      }
+      if (tag['sheet'] === 'dyn' && tag['qt'] === 'premod')
+        return { q: tag['q']! } // Relic stat bonus
+      if (tag['q'] === 'count' && relicSetKeys.has(tag['sheet'] as any))
+        return { q: tag['sheet']! } // Relic set counter
+      if (
+        tag['qt'] == 'lightCone' &&
+        ['lvl', 'ascension', 'superimpose'].includes(tag['q'] as string)
+      )
+        return { q: tag['q']! } // Light cone bonus
+      if (tag['q'] === 'count' && lightConeKeys.has(tag['sheet'] as any))
+        return { q: tag['sheet']! } // Relic set counter
+
+      return undefined
+    })
+    const buildComponents = [
+      this.lightConeStats,
+      // 1-by-1 for deterministic order
+      this.relicStatsBySlot.head,
+      this.relicStatsBySlot.hands,
+      this.relicStatsBySlot.body,
+      this.relicStatsBySlot.feet,
+      this.relicStatsBySlot.sphere,
+      this.relicStatsBySlot.rope,
+    ]
+    const constraints = this.statFilters.map((filter) =>
+      filter.isMax ? filter.value * -1 : filter.value
     )
-    return detachedNodes
+    const {
+      nodes: prunedNodes,
+      candidates,
+      minimum,
+    } = prune(detachedNodes, buildComponents, 'q', constraints, this.topN)
+    this.lightConeStats = candidates[0]
+    // 1-by-1 for deterministic order
+    this.relicStatsBySlot.head = candidates[1]
+    this.relicStatsBySlot.hands = candidates[2]
+    this.relicStatsBySlot.body = candidates[3]
+    this.relicStatsBySlot.feet = candidates[4]
+    this.relicStatsBySlot.sphere = candidates[5]
+    this.relicStatsBySlot.rope = candidates[6]
+    minimum.map((v, i) => (this.statFilters[i].value = v))
```

Comment:
`minimum` may have fewer elements due to `pruning`. You might want to split `statFilter` into `NumTagFree[]` and `number[]` earlier in the setup.